extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js

```diff
@@ -86,28 +86,63 @@ ve.init.wikia.ViewPageTarget.static.actionsToolbarConfig = [
 	}
 ];
 
+ve.init.wikia.ViewPageTarget.prototype.onSurfaceReady = function () {
+	// Parent method
+	ve.init.mw.ViewPageTarget.prototype.onSurfaceReady.call( this );
+
+	if ( !this.anonWarningWidget ) {
+		this.anonWarningWidget = new ve.ui.WikiaAnonWarningWidget();
+		this.anonWarningWidget.setupAnonWarning( this.getToolbar() );
+		this.anonWarningWidget.connect( this, {
+			logInSuccess: 'onLogInSuccess'
+		} );
+	}
+
+	if ( !this.licenseWidget ) {
+		this.licenseWidget = new ve.ui.WikiaLicenseWidget();
+		this.licenseWidget.setupLicense( "#WikiaArticle" );
+	}
+
+};
+
 /**
  * @inheritdoc
  */
-ve.init.wikia.ViewPageTarget.prototype.setSurface = function ( surface ) {
+ve.init.wikia.ViewPageTarget.prototype.onSave = function (
+	html, categoriesHtml, newid, isRedirect, displayTitle, lastModified, contentSub
+) {
+	this.tearDownLicense();
+	this.tearDownAnonWarning();
+
 	// Parent method
-	ve.init.mw.ViewPageTarget.prototype.setSurface.call( this, surface );
-	this.setupLicense( surface );
+	ve.init.mw.ViewPageTarget.prototype.onSave.call( this, html, categoriesHtml, newid, isRedirect, displayTitle, lastModified, contentSub );
 };
 
 /**
  * @inheritdoc
  */
 ve.init.wikia.ViewPageTarget.prototype.tearDownSurface = function ( noAnimate ) {
-	if ( this.$license ) {
-		this.$license.remove();
-		this.$license = null;
-	}
+	this.tearDownLicense();
+	this.tearDownAnonWarning();
 
 	// Parent method
 	return ve.init.mw.ViewPageTarget.prototype.tearDownSurface.call( this, noAnimate );
 };
 
+ve.init.wikia.ViewPageTarget.prototype.tearDownLicense = function () {
+	if ( this.licenseWidget ) {
+		this.licenseWidget.removeLicense();
+		this.licenseWidget = null;
+	}
+};
+
+ve.init.wikia.ViewPageTarget.prototype.tearDownAnonWarning = function () {
+	if ( this.anonWarningWidget ) {
+		this.anonWarningWidget.removeAnonWarning();
+		this.anonWarningWidget = null;
+	}
+};
+
 //ve.init.wikia.ViewPageTarget.static.surfaceCommands.push( 'wikiaSourceMode' );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)